### PR TITLE
wg-naming: Add workflow for proposing recommendations

### PR DESCRIPTION
**Pull request by justaugustus**
Rough draft of the workflow to propose naming recommendations to @kubernetes/wg-naming.

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

(Note that this will eventually link out to content from https://github.com/kubernetes/community/pull/5206 and https://github.com/kubernetes/community/pull/5207, so certain links may be broken until they merge.)

/wg naming

### CODE REVIEW

File: wg-naming/workflow.md
Comment (zacharysarah) [inline wg-naming/workflow.md:82]:
```suggestion
- ADR is on hold until approved by scoped areas (SIG Architecture, SIG Docs)
```

File: wg-naming/workflow.md
Comment (zacharysarah) [inline wg-naming/workflow.md:83]:
```suggestion
- WG Naming lead merges ADR
```

File: wg-naming/workflow.md
Comment (zacharysarah) [inline wg-naming/workflow.md:84]:
```suggestion
- WG Naming lead proposes ADR to Steering for official approval
```

File: wg-naming/workflow.md
Comment (zacharysarah) [inline wg-naming/workflow.md:89]:
```suggestion
- WG Naming leads record accepted recommendation in a canonical location (TBD)
```

File: wg-naming/workflow.md
Comment (zacharysarah) [inline wg-naming/workflow.md:91]:
```suggestion
- Areas in scope are now responsible for implementation
```

**Review by zacharysarah — CHANGES_REQUESTED** [review]
@justaugustus Thank you for this! ✨  Some edits, mostly to eliminate passive voice and increase clarity.

**Comment by justaugustus** [comment]
Thanks for the review, @zacharysarah! Lots of helpful tips :two_hearts: 

@kubernetes/wg-naming -- I'm setting a lazy consensus date of Wednesday, 10/28 at 5pm US Eastern for any final comments from the WG.
After that, let's forward this along to Steering for review.

**Comment by justaugustus** [comment]
(Fixing OWNERS file in https://github.com/kubernetes/community/pull/5270.)

**Comment by justaugustus** [comment]
> I'm setting a lazy consensus date of Wednesday, 10/28 at 5pm US Eastern for any final comments from the WG.

We've cleared the WG lazy consensus period here, so I'd like to open this to review from @kubernetes/steering-committee.
/assign @mrbobbytables @parispittman

**Comment by mrbobbytables** [comment]
/hold
Just for full steering review 👍

**Review by celestehorgan — CHANGES_REQUESTED** [review]
Just a few wording corrections.

Once implemented this has my LGTM.

**Comment by dims** [comment]
LGTM : +1 to ship and iterate!

File: wg-naming/workflow.md
Comment (nikhita) [inline wg-naming/workflow.md:78]: I'm probably missing something here, so please feel free to point me to a link/doc if this was discussed before :)

What would be the difference between merge vs "official approval"?

Comment (dims) [reply wg-naming/workflow.md:78]: I'd say skip 3 and 4. just tag steering, do a lazy consensus timeout and merge.

**Comment by mrbobbytables** [comment]
Overall looks good :)
@celestehorgan's are good, but I do agree with @dims that we can safely merge and iterate. The approval process is flagged as still under discussion, so we can adjust the process to iron out any kinks going forward.

File: wg-naming/workflow.md
Comment (liggitt) [inline wg-naming/workflow.md:71]: we might want to clarify this point... "everything isn’t wholly figured out" is really vague :-)

I'd suggest something specific like:

> Don’t block recording a recommendation on a plan to remediate all existing uses; once the direction is agreed on by the code owners from the scoped area, a recorded recommendation has value in guiding new/future work

**Review by liggitt — COMMENTED** [review]
One suggestion for clarifying the "guiding future work" point. Looks like a good starting point otherwise.

One thing to keep in mind is [long-term ownership of this process](https://github.com/kubernetes/community/tree/master/wg-naming#goals), and thinking about homes for these records/discussions/processes that aren't inextricably tied to `wg-naming` repos/mailing lists. The earlier we think about that, the fewer links we'll accumulate we'll need to chase later and change.

**Comment by justaugustus** [comment]
Thanks for the review, Steering folks!
This is ready for approval now.

---

> I'm probably missing something here, so please feel free to point me to a link/doc if this was discussed before :)
> 
> What would be the difference between merge vs "official approval"?

@nikhita -- We had played with the idea of batching requests for Steering approval, but the regular approval flow works well too. :)

---

> I'd say skip 3 and 4. just tag steering, do a lazy consensus timeout and merge.

@dims -- Updated the language in https://github.com/kubernetes/community/pull/5211/commits/fcb5aa7c61597784de13e2ba81ec81c6872002b9 to:

> - ADR is on hold until approved by scoped areas (e.g., SIG Architecture, SIG
>  Docs)
> - Steering is tagged on the ADR for approval
> - WG Naming lead establishes a Steering review/approval period with a lazy
>   consensus deadline of 3-5 business days
> - WG Naming lead releases the hold and merges ADR

---

> One suggestion for clarifying the "guiding future work" point. Looks like a good starting point otherwise.
> 
> One thing to keep in mind is [long-term ownership of this process](https://github.com/kubernetes/community/tree/master/wg-naming#goals), and thinking about homes for these records/discussions/processes that aren't inextricably tied to `wg-naming` repos/mailing lists. The earlier we think about that, the fewer links we'll accumulate we'll need to chase later and change.

@liggitt -- Alluded to in the `Implementation` section, but we can start to refine that in the next revision.

> - WG Naming leads record accepted recommendation in a canonical location (TBD)
>   (for example, a style guide)
> - Areas in scope are now responsible for implementation

---

> we might want to clarify this point... "everything isn’t wholly figured out" is really vague :-)
> 
> I'd suggest something specific like:
> 
> > Don’t block recording a recommendation on a plan to remediate all existing uses; once the direction is agreed on by the code owners from the scoped area, a recorded recommendation has value in guiding new/future work

@liggitt -- Added this language in https://github.com/kubernetes/community/commit/fcb5aa7c61597784de13e2ba81ec81c6872002b9.

**Comment by liggitt** [comment]
> @liggitt -- Added this language in [fcb5aa7](https://github.com/kubernetes/community/commit/fcb5aa7c61597784de13e2ba81ec81c6872002b9).

thanks, lgtm

**Comment by justaugustus** [comment]
And away we go! Thanks again for the reviews, everyone.
Slack thread to close the loop is here: https://kubernetes.slack.com/archives/C016DM451K4/p1604525808062300

/hold cancel

**Pull request merged**
